src/Sylius/Bundle/OrderBundle/Controller/OrderItemController.php

```diff
@@ -42,12 +54,186 @@ public function createNew()
     }
 
     /**
-     * Get order repository.
+     * @param Request $request
      *
+     * @return Response
+     */
+    public function addAction(Request $request)
+    {
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
+
+        $this->isGrantedOr403($configuration, ResourceActions::CREATE);
+        $newResource = $this->newResourceFactory->create($configuration, $this->factory);
+
+        $this->getItemQuantityModifier()->modify($newResource, 1);
+
+        $form = $this->resourceFormFactory->create($configuration, $newResource);
+
+        if ($request->isMethod('POST') && $form->submit($request)->isValid()) {
+            $newResource = $form->getData();
+
+            $event = $this->eventDispatcher->dispatchPreEvent('sylius.cart_item.pre_create', $configuration, $newResource);
+
+            if ($event->isStopped() && !$configuration->isHtmlRequest()) {
+                throw new HttpException($event->getErrorCode(), $event->getMessage());
+            }
+            if ($event->isStopped()) {
+                $this->flashHelper->addFlashFromEvent($configuration, $event);
+
+                return $this->redirectHandler->redirectToIndex($configuration, $newResource);
+            }
+
+            $cart = $this->getCurrentCart();
```

Comment:
Not sure about it - we should consider naming, as in some points **Cart** is still more appropriate than **Order** - which is now means by ``Order`` in ``cart`` state, not by other entity.